packages/service-library/src/data-providers/website-scan-result-provider.ts

```diff
@@ -54,29 +69,43 @@ export class WebsiteScanResultProvider {
      * Source document properties that resolve to undefined are skipped if a destination document value exists.
      * Will remove all falsey (false, null, 0, "", undefined, and NaN) values from document's array type properties
      */
-    public async mergeOrCreate(websiteScanResult: Partial<WebsiteScanResult>): Promise<WebsiteScanResult> {
-        const scanResultNormalized = this.normalizeToDbDocument(websiteScanResult);
+    public async mergeOrCreate(scanId: string, websiteScanResult: Partial<WebsiteScanResult>): Promise<WebsiteScanResultBase> {
+        const dbDocument = this.convertToDbDocument(scanId, websiteScanResult);
 
         return (await this.retryHelper.executeWithRetries(
-            async () => {
-                const operationResult = await this.createIfNotExists(scanResultNormalized);
-                if (operationResult.created) {
-                    return operationResult.scanResult;
-                } else {
-                    return this.mergeAndWrite(operationResult.scanResult, websiteScanResult);
-                }
-            },
+            async () => this.mergeOrCreateImpl(dbDocument),
             async (err) =>
                 this.logger.logError(`Failed to update website scan result Cosmos DB document. Retrying on error.`, {
-                    document: JSON.stringify(scanResultNormalized),
+                    baseId: dbDocument.baseDocument.id,
+                    partId: dbDocument?.partDocument.id,
```

Comment:
Should the first line here also have a null check for dbDocument?

Reply:
updated